src/types.cpp

```diff
@@ -541,3 +541,30 @@ const Struct *SizedType::GetStruct() const
 }
 
 } // namespace bpftrace
+
+namespace std {
+size_t hash<bpftrace::SizedType>::operator()(
+    const bpftrace::SizedType &type) const
+{
+  auto hash = std::hash<unsigned>()(static_cast<unsigned>(type.type));
+  bpftrace::hash_combine(hash, type.GetSize());
+
+  if (type.IsIntegerTy())
+    bpftrace::hash_combine(hash, type.IsSigned());
+  else if (type.IsRecordTy())
+    bpftrace::hash_combine(hash, type.GetName());
```

Comment:
Shouldn't this be the hash of `inner_struct_`?

Reply:
The `IsEqual` method compares structs by name, so I followed that behaviour. Both options should be correct as we don't have redefined types, but I have no problem changing this to hash the `inner_struct_` (though I'd change the `IsEqual` method, too). 

The question is if we want to have the same hash for two distinctly named types that have the same inner structure.